### PR TITLE
ReCaptcha checking for user comments

### PR DESCRIPTION
**Pull request by crhallberg**
- [x] Test configuration settings
- [x] Test Javascript with settings off
- [x] Lightbox captchas don't appear if comments captcha is displaying
- [x] `View\Helper\RecordTest::testGetTab`
- [x] Double flash message in 'flash' mode, unhelpful error in 'throw' mode

### CODE REVIEW

File: module/VuFind/src/VuFind/View/Helper/Root/Record.php [hunk 1]
```diff
@@ -380,6 +380,7 @@ public function getTab(\VuFind\RecordTab\TabInterface $tab)
         $classParts = explode('\\', get_class($tab));
         $template = 'RecordTab/' . strtolower(array_pop($classParts)) . '.phtml';
         $oldContext = $this->contextHelper->apply($context);
+        $this->view->recordTab = $tab;
```
Comment (crhallberg) [inline module/VuFind/src/VuFind/View/Helper/Root/Record.php]: This line is breaking phpunit.

File: module/VuFind/src/VuFind/RecordTab/Factory.php [hunk 2]
```diff
@@ -262,6 +262,13 @@ public static function getReviews(ServiceManager $sm)
     public static function getUserComments(ServiceManager $sm)
     {
         $capabilities = $sm->getServiceLocator()->get('VuFind\AccountCapabilities');
-        return new UserComments('enabled' === $capabilities->getCommentSetting());
+        $config = $sm->getServiceLocator()->get('VuFind\Config')->get('config');
+        $useRecaptcha = isset($config->Captcha) && isset($config->Captcha->forms)
+            && (trim($config->Captcha->forms) === '*'
+            || strpos($config->Captcha->forms, 'userComments'));
```
Comment (crhallberg) [inline module/VuFind/src/VuFind/RecordTab/Factory.php]: These three lines are a copy of logic from `/Controller/Plugin/Recaptcha->validate()`. I couldn't find a way to use that code here in the Factory though.

Comment (demiankatz) [inline module/VuFind/src/VuFind/RecordTab/Factory.php]: In theory, you could refactor the controller plugin to expose the setting as a public method, and you could rewrite the factory to fetch the controller plugin out of the service manager and use it... but for four lines of code, it's probably just as easy to leave it alone for now!

**Comment by demiankatz** [comment]
A couple of comments:

1.) I try to avoid using public properties, since it can make future changes harder to implement (if you have a wrapper method, you can add logic or change behavior... if you have a direct property access you're basically stuck). Thus, I've converted the public property you added into protected with an accessor.

2.) I see that you've removed the active() method from the reCAPTCHA view helper. However, this is still being called in the jquerymobile theme, so removing it will break the theme. We might want to investigate the status of jquerymobile recaptcha support.

3.) Looks like Google has dropped support for non-JS users in reCAPTCHA. I'm pretty sure it used to work either way, but now you just get a message telling you to turn on JS. (Not a big deal -- just something I noticed during testing).

Seems to be working fine in the Bootstrap-based themes, but we need to do something about issue 2 above before we can merge this.

**Pull request merged**